language/functional_tests/tests/testsuite/sorted_linked_list/sorted-linked-list.mvir

```diff
@@ -0,0 +1,510 @@
+
+
+        return;
+    }
+
+    //removes the current (non-head) node
+    public remove_node_by_node_owner()  acquires Node
+    {
+        let here: address;
+        here = get_txn_sender();
+
+        //make sure a node exists
+        assert(exists<Node>(copy(here)), 11);
+
+        //make sure it is not a head node (heads can be removed using remove_list)
+        assert(!exists<IsHeadNode>(copy(here)),12);
+
+        //remove it
+        Self.remove_node(move(here));
+
+        return;
+    }
+
+    public remove_node_by_list_owner(node_address: address) acquires Node
+    {
+
+        let node_ref: & Self.Node;
+        let list_owner: address;
+
+        //make sure the node exists
+        assert(exists<Node>(copy(node_address)), 13);
+
+        //make sure it is not a head node
+        assert(!exists<IsHeadNode>(copy(node_address)), 14);
+
+        //make sure the caller owns the list
+        node_ref = borrow_global<Node>(copy(node_address));
+        list_owner = *&move(node_ref).head;
+        assert(move(list_owner) == get_txn_sender(), 15);
+
+        //remove it
+        Self.remove_node(move(node_address));
+
+        return;
+    }
+
+    //private function used for removing a non-head node -- does not check permissions
+    remove_node(node_address: address) acquires Node
+    {
+        let current_node_ref: & Self.Node;
+        let next_node_address: address;
+        let next_node_mut_ref: &mut Self.Node;
+        let prev_node_address: address;
+        let prev_node_mut_ref: &mut Self.Node;
+        let temp_address: address;
+        let temp_value: u64; //TODO: make generic
+
+        //make sure the node exists
+        assert(exists<Node>(copy(node_address)),16);
+
+        //make sure it is not a head node
+        assert(!exists<IsHeadNode>(copy(node_address)),17);
+
+        //find prev and next
+        current_node_ref = borrow_global<Node>(copy(node_address));
+        next_node_address = *&copy(current_node_ref).next;
+        prev_node_address = *&move(current_node_ref).prev;
+
+        //update next
+        next_node_mut_ref = borrow_global_mut<Node>(copy(next_node_address));
+        *&mut move(next_node_mut_ref).prev = copy(prev_node_address);
+
+        //update prev
+        prev_node_mut_ref = borrow_global_mut<Node>(move(prev_node_address));
+        *&mut move(prev_node_mut_ref).next = move(next_node_address);
+
+        //destroy the current node
+        Node {temp_address,temp_address,temp_address,temp_value} = move_from<Node>(move(node_address));
+
+        return;
+    }
+
+
+}
+
+
+//! new-transaction
+//! sender: alice
+//creating a new list _@alice
+import {{sys}}.SortedLinkedList;
+main() {
+    SortedLinkedList.create_new_list();
+    return;
+}
+// check: EXECUTED
+
+//! new-transaction
+//! sender: alice
+//attempting to create another list with the same head
+import {{sys}}.SortedLinkedList;
+main() {
+    SortedLinkedList.create_new_list();
+    return;
+}
+// check: ABORTED
```

Comment:
Check error code here (and elsewhere where there are explicit aborts)?

Reply:
Done